wp-lazy-loading.php

```diff
@@ -118,57 +120,115 @@ function wp_lazy_loading_enabled( $tag_name, $context ) {
 }
 
 /**
- * Add `loading="lazy"` to `img` HTML tags.
- *
- * Currently the "loading" attribute is only supported for `img`, and is enabled by default.
+ * Filters 'img' elements in post content and modifies their markup.
  *
  * @since (TBD)
  *
+ * @see wp_image_add_srcset_and_sizes()
+ * @see wp_image_add_loading()
+ *
  * @param string $content The HTML content to be filtered.
  * @param string $context Optional. Additional context to pass to the filters. Defaults to `current_filter()` when not set.
- * @return string Converted content with 'loading' attributes added to images.
+ * @return string Converted content with images modified.
  */
-function wp_add_lazy_load_attributes( $content, $context = null ) {
+function wp_filter_content_attachment_images( $content, $context = null ) {
```

Comment:
If we are going to deprecate `wp_make_content_images_responsive()` (this is a replacement function for it), thinking that limiting it to only images that have a specific class name would be wrong.

Perhaps something like https://github.com/WordPress/wp-lazy-loading/pull/9/files#diff-229cc76004131ce077176ab5955e8707R236 but limited only to `img` tags would be better? Otherwise the new functions still has a limitation that doesn't seem to make sense.

Reply:
We could easily change the regex to also cover images without that class if we decide to. In that case, we would only try to get the attachment if the class is present, and otherwise assume 0.

Reply:
Yeah, looked a bit at how to do this and didn't like any of the alternatives. As it is currently, we will do `_prime_post_caches()` (expensive) even when not adding srcset and sizes. Can probably make that and `wp_get_attachment_metadata()` conditional on `$add_srcset_sizes`. But then cannot pass `$image_meta` to `wp_image_add_loading()` and passing `$attachment_id` will be "undesirable" as plugins would probably use it to get the meta anyway.

Still thinking moving `wp_image_add_loading()` to the first loop makes more sense. Alternatively we should only run this on `the_content` as it is currently for srcset and sizes.